File: test_data/test_sample.py
# 测试文件示例 - 用于 run_tests function

import unittest

def calculate_total(items):
    """计算总价"""
    total = 0
    for item in items:
        total += item.get('price', 0) * item.get('quantity', 1)
    return total

class TestCalculateTotal(unittest.TestCase):
    
    def test_empty_list(self):
        """测试空列表"""
        self.assertEqual(calculate_total([]), 0)
    
    def test_single_item(self):
        """测试单个商品"""
        items = [{'price': 10, 'quantity': 2}]
        self.assertEqual(calculate_total(items), 20)
    
    def test_multiple_items(self):
        """测试多个商品"""
        items = [
            {'price': 10, 'quantity': 2},
            {'price': 5, 'quantity': 3}
        ]
        self.assertEqual(calculate_total(items), 35)
    
    def test_missing_quantity(self):
        """测试缺少数量字段"""
        items = [{'price': 10}]
        self.assertEqual(calculate_total(items), 10)

if __name__ == '__main__':
    unittest.main()

File: test_data/test_sample.js
// 测试文件示例 - 用于 run_tests function

function calculateTotal(items) {
    let total = 0;
    for (let item of items) {
        total += (item.price || 0) * (item.quantity || 1);
    }
    return total;
}

// 简单的测试框架
function assert(condition, message) {
    if (!condition) {
        throw new Error(`Assertion failed: ${message}`);
    }
}

function test(name, fn) {
    try {
        fn();
        console.log(`✓ ${name}`);
        return true;
    } catch (error) {
        console.error(`✗ ${name}: ${error.message}`);
        return false;
    }
}

// 运行测试
function runTests() {
    let passed = 0;
    let failed = 0;
    
    test('empty list', () => {
        assert(calculateTotal([]) === 0, 'Empty list should return 0');
    }) && passed++ || failed++;
    
    test('single item', () => {
        assert(calculateTotal([{price: 10, quantity: 2}]) === 20, 'Should calculate correctly');
    }) && passed++ || failed++;
    
    test('multiple items', () => {
        assert(calculateTotal([
            {price: 10, quantity: 2},
            {price: 5, quantity: 3}
        ]) === 35, 'Should sum multiple items');
    }) && passed++ || failed++;
    
    console.log(`\nTests: ${passed} passed, ${failed} failed`);
    return failed === 0;
}

if (require.main === module) {
    const success = runTests();
    process.exit(success ? 0 : 1);
}

module.exports = { calculateTotal, runTests };
